src/library/autodj/autodjprocessor.h

```diff
@@ -154,6 +152,14 @@ class AutoDJProcessor : public QObject {
         ADJ_NOT_TWO_DECKS
     };
 
+    enum class TransitionMode {
+        AlignIntroOutroStart = 0,
+        AlignIntroOutroEnd = 1,
```

Comment:
> But unfortunately not in the corner case, where the outro is only 1 second. In this case of the new track are directly placed next to the vocals of the new track.

How long is the intro in this case you are considering?

> Here a AlignIntroOutroStart would be suitable. For unattended mode we need a rule for that.
How about using a AlignIntroOutroStart as long as outro size is < transiotion time?

So are you proposing adding a new mode that automatically switches between AlignIntroOutroStart & AlignIntroOutroEnd?

Reply:
> How long is the intro in this case you are considering? 

In this case 10 s. A radio mix. 

> So are you proposing adding a new mode that automatically switches between AlignIntroOutroStart & AlignIntroOutroEnd?

Yes, and it should actually replace the both others. It is because I cannot decide which mode is the best before starting an unattended run. And even it is attended, I would avoid to swap between these modes. 
Which one did you use lately?

Reply:
I have mostly used the align intro + outro start mode. I would rather have AutoDJ behave predictably with simple rules that sound good enough in most cases than have AutoDJ sound ideal in all cases if the rules are hard to understand or explain. Depending on the particular tracks being mixed, yes, sometimes aligning the start points will sound better than the end points will sound better, or the other way around will sound better. I do not know how Mixxx could determine which would be better automatically, but I hope either is at least good enough in all cases. I am doubtful that any method which relies on an arbitrary length of time from the spinbox will be very good.

Reply:
Maybe what you want is the behavior of the old "Intro + outro (shorter)" mode? You can test that again by checkout out 3971e4a61f84ab69ac36a4da6892bb349dd89d94. When I started this PR, I was thinking about and explaining that in terms of the old AutoDJ behaviors. I think it is clearer to understand it in terms of how the tracks align:
  * If outro > intro, align start points of intro + outro.
  * If intro > outro, align end points of intro + outro.
We could bring this mode back. The question then is what short name should we give it for the mode selection combobox.

Reply:
Please describe your use case a bit more. How long is the intro outro of a typical good working track? Do you shift the intro start an outro end manually?
If yes to which point. Which rules do you use to find this point?
When does the start start alignment fail? Which type of tracks are involved.

The original transition will work well for radio/radio track transitions. It will fail badly if a club tracked is cued. Because it plays always the longest of both phases.

Reply:
> How long is the intro outro of a typical good working track?

There is no typical; it depends on the track. Sometimes it is 2 seconds, sometimes it is 45 seconds.

> Do you shift the intro start an outro end manually?

Only rarely when I want to start nowhere near the beginning of the track. For example I have a long jazz track where I only play the end because an audience would probably get bored by the first 5 minutes of the track:
![image](https://user-images.githubusercontent.com/9455094/59277951-b3851500-8c26-11e9-8466-a755d908ca60.png)
For the vast majority of tracks I leave the intro start at the autodetected first sound.

> The original transition will work well for radio/radio track transitions. It will fail badly if a club tracked is cued. Because it plays always the longest of both phases.

So you want to avoid long crossfades? If so, the behavior of the old "Intro/outro (shorter)" mode described above does that.

Reply:
I want mainly reduce the options, because you can't switch them if you are not at your desk. 
It is nice to read, that we will use the auto DJ quite similar so we should be able to com up with a nice automated solution.

Except of corner cases that are always there, I like the mayority of transition work like this: 
* Phrase synced.
* ~10 sec 
* Transition MUST be finish before vocals start
* Transition must not start before vocals end.
* Boring club intros should be shorten.

Does this match with your ideas?

Reply:
Please try the "Intro/outro (shorter)" mode in 3971e4a. If that works well for you, I can reimplement it.

> Boring club intros should be shorten.

This is not a big deal for me, but I wouldn't mind if it happened.

Reply:
Ping. I am still unsure how to proceed here. Should I reimplement the "Intro/outro (shorter)" mode?

Reply:
No. I need to test It again, to come up with a clear opinion. 

My initial issue is that we have now a set of modes, that are all suitable for some conditions. 
But in a unmaintained run, you may have all conditions. Which mode should the DJ select? 
It took me a while to wrap my head around all the issues here and I am afraid the average user might be lost here. 

I am looking for a solution that can be selected by default and has always reasonable results.